src/pools/abstract-pool.ts

```diff
@@ -152,11 +152,10 @@ export abstract class AbstractPool<
   protected chooseWorker (): Worker {
     if (this.workers.length - 1 === this.nextWorker) {
       this.nextWorker = 0
-      return this.workers[this.nextWorker]
     } else {
       this.nextWorker++
-      return this.workers[this.nextWorker]
     }
+    return this.workers[this.nextWorker]
```

Comment:
You could also try 

```ts
    this.nextWorker =
      this.workers.length - 1 === this.nextWorker ? 0 : this.nextWorker + 1
    return this.workers[this.nextWorker]
```

But I struggle with myself if that's less readable :thinking: 

I personally would prefer another formatting like

```ts
    this.nextWorker = this.workers.length - 1 === this.nextWorker
      ? 0
      : this.nextWorker + 1
    return this.workers[this.nextWorker]
```

but the formatter reformat it back :slightly_frowning_face: 

We could also add a `getter` for the equation so it becomes 

```ts
  protected get aGetterName (): boolean {
    return this.workers.length - 1 === this.nextWorker
  }

  protected chooseWorker (): Worker {
    this.nextWorker = this.aGetterName ? 0 : this.nextWorker + 1
    return this.workers[this.nextWorker]
  }
```

Reply:
Ternary operator are fine.